spanner/client.go

```diff
@@ -477,6 +477,8 @@ func (c *Client) BatchReadOnlyTransaction(ctx context.Context, tb TimestampBound
 	t.txReadOnly.qo = c.qo
 	t.txReadOnly.ro = c.ro
 	t.txReadOnly.disableRouteToLeader = true
+	t.txReadOnly.qo.DirectedReadOptions = c.dro
```

Comment:
@rahul2393 
Need your help to validate this case. Lets discuss on Monday.

Reply:
Directed read options can be set for Partitioned read and Partitioned query since they internally call read and execute_streaming_sql methods. Since the transaction type for `BatchReadOnlyTransaction` is read-only, we need to consider the client level option as fallback.